### DISCUSSION TITLE
Adjusting memory limit

### DISCUSSION THREAD

**Discussion by arcstur** [nextcloud/all-in-one#3143] [Questions] [answered]
According to the [documentation](https://github.com/nextcloud/all-in-one#how-to-adjust-the-php-memory-limit-for-nextcloud), to adjust the memory limit I should stop the master container, remove it, and then start it again with the initial command.

Do I need to also stop and remove the other containers when doing this?

**Comment by szaimen** [comment, marked as answer]
> I should stop the master container, remove it, and then start it again with the initial command.

Yes

> Do I need to also stop and remove the other containers when doing this?

After recreating the mastercontainer simply stop and start the containers again from the aio interface and it should work.

**Comment by Llewellynvdm** [comment]
I also found the memory limit, but changing this value `NEXTCLOUD_MEMORY_LIMIT` did not resolve the issue for me, I had to do the following:

# How to fix the PHP_MAX_CHILDREN limit

This causes the system to freeze/hang when many activity is done on the system. The default seems way to low, (350+-) I bump it to (900) but I think we can even take it higher, depending on your system size.

### You will need access the server!

Open the master container:
```
docker exec -it --user root  nextcloud-aio-mastercontainer /bin/bash
```

Then edit this file:
```
vi php/containers.json
```

Look for the object of the nextcloud:
```
"container_name": "nextcloud-aio-nextcloud"
```

Then add the following mount point (check the last volume /entrypoint.sh):
```
     "volumes": [                                  
        {
          "source": "nextcloud_aio_nextcloud",
          "destination": "/var/www/html",
          "writeable": true
        },
        {
          "source": "%NEXTCLOUD_DATADIR%",
          "destination": "/mnt/ncdata",
          "writeable": true
        },
        {
          "source": "%NEXTCLOUD_MOUNT%",
          "destination": "%NEXTCLOUD_MOUNT%",
          "writeable": true
        },
        {
          "source": "%NEXTCLOUD_TRUSTED_CACERTS_DIR%",
          "destination": "/usr/local/share/ca-certificates",
          "writeable": false
        },
        {
          "source": "/home/ubuntu/nextcloud/entrypoint.sh",
          "destination": "/entrypoint.sh",
          "writeable": true
        }
     ]
```

When done editing the file:
```
press: [Esc]
```

Then to save changes:
```
press: [:w] and [enter]
```

To close vi editor:
```
press: [:q]
```

Check the file on the parent system:
```
/home/ubuntu/nextcloud/entrypoint.sh
```

That it is up to date with:
```
https://github.com/nextcloud/all-in-one/blob/main/Containers/nextcloud/entrypoint.sh
```

But update the following line that says: (Setting php max children...):

Change it to the following:
```
echo "Setting php max children for YOUR SERVER NAME"
# MEMORY=$(awk '/MemTotal/ {printf "%d", $2/1024}' /proc/meminfo)
# PHP_MAX_CHILDREN=$((MEMORY/50))
PHP_MAX_CHILDREN=900
```
You can make it more... if 900 is still not high enough.

Once all that is done, restart the whole system, go to:
```
https://yourserver.com/settings/admin/overview
```

Click on [Open Nextcloud AIO Interface ↗](https://xxx.xx.xx.xx:8080/containers) and **stop** all containers and **start** them all again.

If you can't access `https://yourserver.com/` then go directly to:
```
https://xxx.xx.xx.xx:8080/containers
```
